cmd/soci/commands/ztoc/info.go

```diff
@@ -86,7 +86,7 @@ var infoCommand = cli.Command{
 		if err != nil {
 			return err
 		}
-		gzInfo, err := compression.NewGzipZinfo(ztoc.CompressionInfo.Checkpoints)
+		gzInfo, err := compression.NewZinfo(compression.CompressionGzip, ztoc.CompressionInfo.Checkpoints)
```

Comment:
Based on discussion, `CompressionInfo` is still necessary for now as it holds not only the `Zinfo` but also some metadata such as `SpanDigests` (used for span verification). But there are some optimizations we should do:

1. The `CompressionInfo` should hold a `Zinfo` struct, instead of `Zinfo`'s serilized [bytes](https://github.com/awslabs/soci-snapshotter/blob/878e7d9fbdb67fd030f8ac1b36777a83cd525aea/ztoc/ztoc.go#L59). It doesn't make sense for `Ztoc` to hold structs for other fields but serilized bytes for `Zinfo`. We should just serilize the `Zinfo` struct when necessary(e.g. when we serilize `ztoc`)
2. Span manager should only hold a `ztoc` (not [both `ztoc` and `zinfo`](https://github.com/awslabs/soci-snapshotter/blob/878e7d9fbdb67fd030f8ac1b36777a83cd525aea/fs/span-manager/span_manager.go#L122-L125)): zinfo is part of ztoc. 

I think this can be done in a separate PR as they are additions to this one.